### PR TITLE
Walls made entirely of wood!

### PR DESCRIPTION
**Pull request by fludd12**
:cl: fludd12
add:You can now make wooden walls out of wooden barricades. They're weaker than regular wooden walls, though.
/:cl:

Barricades already look kinda like girders, and making walls entirely out of wood fits with the whole paper wall thing that was added a while back anyway.

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
didnt know space station meant mmorpg:elves dlc

**Comment by JStheguy** [comment]
Well, lavaland IS basically an out of place fantasy "dark land of evil and darkness and ungood", might as well throw in some more fantasy elements.

**Comment by KorPhaeron** [comment]
Lifebringers at least were scifi though

**Comment by Repukan** [comment]
Can't wait for lifebringers to ghost map a way to the station and go over with their new environment deleter staff

**Comment by Repukan** [comment]
Environmental danger deleter, soz

**Comment by 81Denton** [comment]
>Can't wait for lifebringers to ghost map a way to the station and go over with their new environment deleter staff

You could always do a PR where lifebringers aren't allowed to leave lavaland

**Comment by Repukan** [comment]
Or we could not add a way to delete one of the z levels stationary dangers with no effort or input in their part since that's usually locked behind killing a permanently round ending monster?

**Comment by Therapist1** [comment]
i see no problems with it, but if you’re concerned about such things, barring them from the station will work just fine.

**Comment by Therapist1** [comment]
and if you’re concerned about effort, maybe make it require fuel of some kind, or require a lot of rare plants to craft.

**Comment by arsserpentarium** [comment]
About effort, I think, it's fair question. Because it requires very tiny amounts of gaia.
Simple way is to make staff require much more gaia to make and have limited charges.
More interesting and difficult way--to make it require reagents as fuel.

**Comment by Repukan** [comment]
Maybe make it take earth blood/Gaia extract to charge?

**Comment by ma44** [comment]
Wouldn't atmos atmosphere negate the gases produced or removed by the sapling?

**Comment by ExcessiveUseOfCobblestone** [comment]
"throws lifebringers a bone" You mean the individuals that spam kudzu and tomato spam legion?

@Repukan I don't think goliaths are typically round ending to people who spam killer tomatoes and explosive plants but alright.

**Comment by Repukan** [comment]
Meant the lava.

You know...... environment....hazard

**Comment by ExcessiveUseOfCobblestone** [comment]
>  that's usually locked behind killing a permanently round ending monster

A few goliaths* ...

**Comment by Repukan** [comment]
But a Goliath bone boat doesn't make the lava go away.

I was referring to Ash drakes and their staff of lava.

**Comment by Kromgar** [comment]
I think instead of a staff it should be some sort of fruit that releases some sort of terraforming gas.

**Comment by Hatfish** [comment]
just print an rcd, it's so fucking easy to get rid of lava if you don't want it there it's not even funny

**Comment by fludd12** [comment]
@Repukan The RCD can build floors over lava, it has never been a permanent fauna-locked hazard.

@Kromgar I have an idea in mind, will make it run off reagents and act slightly more sciency. Still, saying that hydroponics is purely science is a little silly what with holymelons and earthsblood blessing.

**Comment by Repukan** [comment]
What the fug

**Comment by fludd12** [comment]
Staff now uses earthsblood as fuel, and does not autodelete on use. Values can be changed easily.
Staff can now also shatter rock walls, and plant grass where they were. Getting a lighting issue when I do this, unsure if it'll actually appear in game or not.

**Comment by ExcessiveUseOfCobblestone** [comment]
If you're going to give lifebringers some sort of """endgame""" content I'd recommend removing a lot of the luxuries (specifically the perfect gene machine) because It's pretty easy to reagent spam with botany when you can reliably get the best products after a couple minutes of pulling genes from roundstart seeds.

**Comment by KorPhaeron** [comment]
This shoulda been two separate PRs and I'd prefer they get something like the GECK or Genesis Device they have to fuel with biomass rather than staff

**Comment by arsserpentarium** [comment]
It isn't endgame stuff.
Also biotechnological GEKK is more fit lifebringer theme.

**Comment by fludd12** [comment]
@KorPhaeron The main reason I didn't make it two separate PRs is because I kinda considered the whole wooden wall thing a necessity for the gold bush to work. If you'd like, though, I can simply make this add wooden walls, and maybe rework the gaia blessing to something a bit more thematic at a later date.

**Comment by ExcessiveUseOfCobblestone** [comment]
@arsserpentarium being able to null the map gating for EVERYONE because haha im a plant certainly isn't (shouldn't) be roundstart, which you can do easily when you're given saltpetre (no downside chem that buffs yield and pot) and mutagen in a chem dispenser that doesn't even require you make the chem (saving you energy which the dispenser has plenty of).

Best part about this is once you get it for literally 1 plant you can transfer it across every plant you make with ease with 0 gating. How wild is that?!!

**Comment by fludd12** [comment]
Golems get RCDs and plastitanium shells, Ashwalkers aren't intended to be super advanced but I think they *also* get a RCD, on top of their lava boats and swarm tactics. Currently, podpeople are the only ones who don't have a reasonable method of obtaining safe-ish lava travel. If fifty units is too small to replicate the effects of a RCD once, I can change that, but I do think this is a necessity, @ExcessiveUseOfCobblestone.

**Comment by KorPhaeron** [comment]
@fludd12 It's fine if you want to keep it as one PR, but the wall change could be a 24 hour merge and the rest of it sounds like its gonna be more arguing

**Comment by fludd12** [comment]
I think I'll keep this together, then, and if arguments go on for too much longer, I'll split it. With the freeze coming up, just wanted to crank out a few smaller features while waiting on review for the slime PR, then hopefully do a bunch of fixes.

**Comment by ExcessiveUseOfCobblestone** [comment]
@arsserpentarium lifebringers don't start with one.

If miners want to remove the gating for everyone else that's fine, that dynamic shouldn't be flipped though.

**Comment by ExcessiveUseOfCobblestone** [comment]
Lettuce shift away from my endless conquest against ghost-role second-life syndrome because that boils down to opinions of what a ghost role should/shouldn't be to something a bit more sane, which is that this is PR'd as giving lifebringers something to do that isn't hunt megafauna which it infact just enables them to hunt megafauna better.

**Comment by fludd12** [comment]
Hm. I never actually considered it from that perspective - Whenever I go lifebringer, I tend to just plant grass and glowberries outside, spreading more life, etc. I'm definitely an outlier, though, as I tended to play pacifistic even before the trait was available on bagil. Any suggestions for limiting stuff to that effect while still preserving the ability to "purify" the environment?

**Comment by ExcessiveUseOfCobblestone** [comment]
@fludd12 well of course!

Perhaps a special seed for lifebringers that grows into a "Lifetree" (can only grow in dirt, hydroponic trays would break due to its weight) that, when fully grown, slowly does what the lifestaff does but randomly and slowly around an area (think cult pylon but NATURE and slower). sometimes they spawn butterflies 🦋 

If it dies though it instead turns the area into asteroid, chasms, and lava + shades or something evil until you remove it so you can't just use it to heck off, you have to maintain it or come back to a nasty surprise.

obv needs a check for gaia dirt piles

**Comment by fludd12** [comment]
This actually sounds like an idea I had *quite* a long time ago, which I could modify to fit with this. Maybe, instead of requiring maintenance and such, you make a large tree that you can *enter* as a podperson, putting you in control of the tree and letting you control root/grass growth much like a blob. Sounds like a horrifyingly complex idea to code, so I might do that after both the slime rework and the freeze, but I like that idea a lot better than this one. Switching to just adding the walls here.

**Comment by ExcessiveUseOfCobblestone** [comment]
funny enough I've had a similar Idea as well but as a separate ghost role from planting a seed in hydroponics that could empower plants (this was before the gene machine and even bees existed, way before I knew how to do much of anything codewise)

**Comment by KingPhilipIII** [comment]
I like the idea of having a definite goal to work towards but I feel like calling it a round end thing gives it a bit more weight than it really holds. I doubt this will be so abused like people seem to think, excluding immediately after it's merged(Must try new thing to max). 

I do understand where the opposition is coming for, I just disagree on the exigence of your argument.
On the record though, they were originally just a botany tutorial and an end-goal was never the idea.

**Comment by ExcessiveUseOfCobblestone** [comment]
The individuals who use the role to kill bosses would most certainly use this to navigate to said boss come on now.

It wouldn't be abused by the people who are just there to do the botany tutorial which thankfully outweigh the second-lifers but still.

That said, I think the approach fludd has in mind is better than the original staff idea.

**Comment by arsserpentarium** [comment]
No, that's not fun. Original idea was questionable in details, but it was way more cool.

**Comment by KingPhilipIII** [comment]
I mean. Really can't deny that one cobby.

**Comment by fludd12** [comment]
The idea I have is similar in programming scope to the first part of my slime PR. For the time being, just adding wooden walls is what I have in mind. Currently working on a design document for my lifetree idea, compositing your own idea with one I had for paradise two years ago, @ExcessiveUseOfCobblestone.

Don't worry, everyone. My curse of having too-complex-ideas and maybe-able-to-actually-code them is still a thing, and this idea might turn out *really* fun.

**Comment by ExcessiveUseOfCobblestone** [comment]
"A staff you click on tiles to turn it into grass and water is cooler than becoming a powerscaling nature-being"

What the actual **heck** is this chat?

**Comment by KingPhilipIII** [comment]
No. I meant that it would be abused. Not what's his faces comment.

**Comment by Doppeltongue** [comment]
How about a way to make airless areas full of atmosphere through botany? For instance: maybe some kind of bubble-fruit that passively generates air in low-pressure environments while planted and whose produce expand into tile-sized bubbles when used that blocks atmos changes like tiny fans/holo firewalls? The only plant that currently does this is kudzu, and that's difficult to use even for me. This way, plantpeople have a way to slowly actually terraform lavaland by planting large gardens.

**Comment by fludd12** [comment]
That was actually already a part of the original design for Gaia's Blessing, but I am moving that functionality to the Lifetree at the time in the future I actually get around to coding it. Speaking of which, here is my currently work in progress design document, if you'd like to watch me write up the info about it:

https://docs.google.com/document/d/1FD018NB1dM3FGnN8YeFhqQQwu8zH4N63dIKFq0l3IJo/edit?usp=sharing

**Comment by ExcessiveUseOfCobblestone** [comment]
> botany is now a better atmos

We should honestly have a station build entirely around botany for the (not) meme that botany can do everyone else's job better.

No medical, just hydro trays with poppies and grapes.
Replaces engineering with glowcaps and towercap trays and this new suggestion
Security has nettle trays
Science has xenobio and lemon nades
Service has bananas and moonflowers
Cargo ~~has cashlemons~~ doesn't exist because you have everything you could want with botany.

**Comment by fludd12** [comment]
Actually, a part of the lifetree thing in my design is that they can eventually become what amounts to a station on their own if they are given long enough, but they're fragile and killing the lifetree will destroy everything it has made.

**Comment by optimumtact** [comment]
this game has always had fantastical elements to the sci-fi

Think D&D in space

**Comment by fludd12** [comment]
Regardless, though, that's not what this PR is about! This is now about wooden walls. Will maybe do the special tree in a later PR, but this is done.

File: code/game/turfs/simulated/wall/mineral_walls.dm [hunk 1]
```diff
@@ -130,7 +130,21 @@
...
+	canSmoothWith = list(/turf/closed/wall/mineral/wood, /obj/structure/falsewall/wood, /turf/closed/wall/mineral/wood/nonmetal)
+
+/turf/closed/wall/mineral/wood/nonmetal
+	desc = "A solidly wooden wall. It's a bit weaker than a wall made with metal."
+	girder_type = /obj/structure/barricade/wooden
+	hardness = 50
+	canSmoothWith = list(/turf/closed/wall/mineral/wood, /obj/structure/falsewall/wood, /turf/closed/wall/mineral/wood/nonmetal)
+
+/turf/closed/wall/mineral/wood/nonmetal/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/hatchet) || istype(W, /obj/item/twohanded/fireaxe))
```
Comment (ShizCalev) [inline code/game/turfs/simulated/wall/mineral_walls.dm]: #31126
if you could put this on the parent instead i'd love you

**Comment by KorPhaeron** [comment]
@fludd12

You are welcome to work on the other half of this during the freeze if you want by the way, it was open prior before I asked you to split it

File: code/game/turfs/simulated/wall/mineral_walls.dm (hunk 1 above)
Comment (fludd12) [reply code/game/turfs/simulated/wall/mineral_walls.dm]: Will do!

**Comment by fludd12** [comment]
We'll see if I can even get to that - I still have another 12 slime extract types to get to.

**Comment by fludd12** [comment]
@KorPhaeron This has been merge ready for a while, by the way. Just pinging you in case you forgot about it!

**Pull request merged**